### PR TITLE
feature(overwriteSource): add flag to overwrite source files

### PR DESCRIPTION
**Pull request by code-forger**
Addresses #61

I will write unit tests for this PR once the approach I have used is approved. Let me know if you have any thoughts @coderaiser or if you are happy enough for me to write some tests

### CODE REVIEW

**Comment by coderaiser** [comment]
Would be amazing if you use [yargs-parser](https://www.npmjs.com/package/yargs-parser) for this purpose and move all logic to `lib/cli.js` and add tests with one assertion per test in file `lib/cli.spec.js`, and it can be `ESM`, would be awesome if it will :)

**Comment by code-forger** [comment]
> Would be amazing if you use [yargs-parser](https://www.npmjs.com/package/yargs-parser) for this purpose and move all logic to `lib/cli.js` and add tests with one assertion per test in file `lib/cli.spec.js`, and it can be `ESM`, would be awesome if it will :)

Done!

File: bin/minify.js [hunk 1]
```diff
@@ -12,8 +14,8 @@ const log = function(...args) {
 };
 
 const Argv = process.argv;
-const files = Argv.slice(2);
-const [In] = files;
+const args = yargsParser(Argv.slice(2));
```
Comment (coderaiser) [inline bin/minify.js:17]: Could you please put `yargs-parser` to `lib/cli.js` similar to [yargs-parse example](https://github.com/coderaiser/putout/blob/master/packages/putout/lib/cli/index.js#L82-L139)

File: bin/minify.js [hunk 2]
```diff
@@ -79,15 +83,26 @@ async function processStream(chunks) {
...
 }
 
-function uglifyFiles(files, options) {
+function uglifyFiles(files, options, flags) {
     const minify = require('..');
     const minifiers = files.map((file) => minify(file, options));
     
     Promise.all(minifiers)
-        .then(logAll)
+        .then(async (minifiedStrings) => {
```
Comment (coderaiser) [inline bin/minify.js:91]: Looks like we can use:

```js
await Promise.all(minifiers);
```

And no need in `Promise chain` :)

**Comment by coderaiser** [comment]
Any progress on this?

**Pull request closed without merging**